client/src/main/java/tds/session/Extern.java

```diff
@@ -0,0 +1,45 @@
+package tds.session;
+
+/**
+ * configuration properties for the session system
+ */
+public class Extern {
+    private String clientName;
+    private String environment;
+
+    public static class Builder {
```

Comment:
There isn't a benefit here.  I could've had a constructor that took in the two properties.  The key takeaway with this object is that it is Immutable.  I am using the builder in Exam because there are more than 5 fields and the constructor would be much.

I'm not sure we need our objects to be immutable but I saw no reason why they shouldn't be.